File: wfm-develop mobile 07.05.2026/wfm/mobile/ios/WFMApp/Features/ManagerTasksFlow/TaskFiltersBottomSheet.swift
import SwiftUI
import WFMUI

/// Модель элемента фильтра для UI
struct TaskFilterItem: Identifiable, Equatable {
    let id: String
    let title: String
    var isSelected: Bool
    var isEnabled: Bool = true
}

/// Группа фильтров для UI
struct TaskFilterGroup: Identifiable, Equatable {
    let id: String
    let title: String
    var items: [TaskFilterItem]
}

private let filterChipThreshold = 10

/// BottomSheet фильтров задач (тип работ, сотрудники, зоны)
///
/// Использование:
/// ```swift
/// @EnvironmentObject private var container: DependencyContainer
///
/// Button("Фильтры") {
///     TaskFiltersBottomSheet.show(
///         bottomSheetManager: container.bottomSheetManager,
///         filterGroups: viewModel.filterGroups,
///         onApply: { groups in viewModel.applyFilters(groups) }
///     )
/// }
/// ```
struct TaskFiltersBottomSheet {
    /// Показать BottomSheet фильтров задач
    @MainActor
    static func show(
        bottomSheetManager: BottomSheetManager,
        filterGroups: [TaskFilterGroup],
        taskFilterIndices: [[Int]] = [],
        onApply: @escaping ([TaskFilterGroup]) -> Void
    ) {
        bottomSheetManager.show(showOverlay: true) {
            TaskFiltersBSContent(
                initialGroups: filterGroups,
                taskFilterIndices: taskFilterIndices,
                onApply: onApply,
                onDismiss: { bottomSheetManager.dismiss() }
            )
        }
    }
}

// MARK: - State Wrapper

/// Обёртка с внутренним @State для показа через bottomSheetManager (AnyView не поддерживает внешние @Binding)
private struct TaskFiltersBSContent: View {
    @State private var filterGroups: [TaskFilterGroup]
    let taskFilterIndices: [[Int]]
    let onApply: ([TaskFilterGroup]) -> Void
    let onDismiss: () -> Void

    init(initialGroups: [TaskFilterGroup], taskFilterIndices: [[Int]], onApply: @escaping ([TaskFilterGroup]) -> Void, onDismiss: @escaping () -> Void) {
        _filterGroups = State(initialValue: initialGroups)
        self.taskFilterIndices = taskFilterIndices
        self.onApply = onApply
        self.onDismiss = onDismiss
    }

    var body: some View {
        TaskFiltersBottomSheetContent(
            filterGroups: $filterGroups,
            taskFilterIndices: taskFilterIndices,
            onApply: onApply,
            onDismiss: onDismiss
        )
    }
}

// MARK: - Content

/// Контент BottomSheet фильтров задач
struct TaskFiltersBottomSheetContent: View {
    @Environment(\.wfmColors) private var colors

    @Binding var filterGroups: [TaskFilterGroup]
    let taskFilterIndices: [[Int]]
    let onApply: ([TaskFilterGroup]) -> Void
    let onDismiss: () -> Void

    @State private var expandedGroupId: String? = nil

    private var hasAnySelected: Bool {
        filterGroups.contains { $0.items.contains { $0.isSelected } }
    }

    /// Количество задач, соответствующих текущему выбору (nil если ничего не выбрано)
    private var matchingTaskCount: Int? {
        guard hasAnySelected, !taskFilterIndices.isEmpty else { return nil }
        let activeSelections: [Int: Set<Int>] = filterGroups.enumerated().reduce(into: [:]) { result, pair in
            let (g, group) = pair
            let selected = Set(group.items.indices.filter { group.items[$0].isSelected })
            if !selected.isEmpty { result[g] = selected }
        }
        return taskFilterIndices.filter { row in
            activeSelections.allSatisfy { g, selSet in g < row.count && selSet.contains(row[g]) }
        }.count
    }

    private var showButtonText: String {
        guard let count = matchingTaskCount else { return "Показать задачи" }
        return "Показать \(count) \(taskCountWord(count))"
    }

    private func taskCountWord(_ count: Int) -> String {
        let mod100 = count % 100
        let mod10 = count % 10
        if (11...14).contains(mod100) { return "задач" }
        switch mod10 {
        case 1: return "задачу"
        case 2, 3, 4: return "задачи"
        default: return "задач"
        }
    }

    private var selectedGroupFilters: [(groupId: String, groupTitle: String)] {
        filterGroups.compactMap { group in
            guard group.items.contains(where: { $0.isSelected }) else { return nil }
            return (group.id, group.title)
        }
    }

    var body: some View {
        VStack(spacing: 0) {
            // Заголовок
            Text("Фильтры")
                .font(WFMTypography.headline20Bold)
                .foregroundStyle(colors.textPrimary)
                .frame(maxWidth: .infinity, minHeight: 44, alignment: .leading)
                .padding(.horizontal, WFMSpacing.l)
                .padding(.top, WFMSpacing.m)
                .padding(.bottom, WFMSpacing.s)

            // Секция "Вы выбрали" — показывается если хоть что-то выбрано
            if !selectedGroupFilters.isEmpty {
                selectedTagsSection
                    .padding(.horizontal, WFMSpacing.l)
                    .padding(.bottom, WFMSpacing.m)
            }

            // Аккордеон секций
            VStack(spacing: 0) {
                ForEach(Array(filterGroups.enumerated()), id: \.element.id) { index, group in
                    filterSection(group: group)

                    if index < filterGroups.count - 1 {
                        Divider()
                            .padding(.horizontal, WFMSpacing.l)
                    }
                }
            }

            Divider()

            // Кнопки внизу
            bottomButtons
        }
    }

    // MARK: - Selected Tags

    private var selectedTagsSection: some View {
        VStack(alignment: .leading, spacing: WFMSpacing.s) {
            Text("Вы выбрали")
                .font(WFMTypography.headline16Bold)
                .foregroundStyle(colors.textPrimary)
                .frame(maxWidth: .infinity, minHeight: 40, alignment: .leading)

            FiltersFlowLayout(spacing: WFMSpacing.s) {
                ForEach(selectedGroupFilters, id: \.groupId) { filter in
                    WFMFilterTag(
                        text: filter.groupTitle,
                        onRemove: { deselectAllInGroup(groupId: filter.groupId) }
                    )
                }
            }
        }
        .frame(maxWidth: .infinity, alignment: .leading)
    }

    // MARK: - Filter Section (аккордеон)

    @ViewBuilder
    private func filterSection(group: TaskFilterGroup) -> some View {
        let isExpanded = expandedGroupId == group.id
        let selectedCount = group.items.filter { $0.isSelected }.count

        VStack(alignment: .leading, spacing: 0) {
            // Заголовок-кнопка аккордеона
            Button {
                withAnimation(.easeInOut(duration: 0.2)) {
                    if isExpanded {
                        expandedGroupId = nil
                    } else {
                        // Пересчитываем enabled перед открытием секции
                        filterGroups = recomputeFilterEnabledState(
                            filterGroups: filterGroups,
                            taskFilterIndices: taskFilterIndices
                        )
                        expandedGroupId = group.id
                    }
                }
            } label: {
                HStack(spacing: WFMSpacing.xxs) {
                    Text(group.title)
                        .font(WFMTypography.headline16Bold)
                        .foregroundStyle(colors.textPrimary)

                    if selectedCount > 0 {
                        WFMBadge(text: "\(selectedCount)", color: .violet)
                    }

                    Spacer()

                    if selectedCount > 0 {
                        Button("Очистить") {
                            deselectAllInGroup(groupId: group.id)
                        }
                        .font(WFMTypography.headline12Medium)
                        .foregroundStyle(colors.buttonLinkTextDefault)
                    }

                    (isExpanded ? WFMIcons.chevronUp : WFMIcons.chevronDown)
                        .resizable()
                        .renderingMode(.template)
                        .foregroundStyle(colors.iconPrimary)
                        .frame(width: 20, height: 20)
                }
                .padding(.horizontal, WFMSpacing.l)
                .padding(.vertical, WFMSpacing.s)
                .contentShape(Rectangle())
            }
            .buttonStyle(.plain)

            // Контент: чипы (≤10) или список с чекбоксами (>10)
            if isExpanded {
                if group.items.count <= filterChipThreshold {
                    chipContent(group: group)
                } else {
                    checkboxContent(group: group)
                }
            }
        }
    }

    // Чипы для секций с ≤10 элементами
    private func chipContent(group: TaskFilterGroup) -> some View {
        FiltersFlowLayout(spacing: WFMSpacing.s) {
            ForEach(group.items) { item in
                WFMChip(
                    text: item.title,
                    state: !item.isEnabled ? .disabled : item.isSelected ? .active : .default,
                    onTap: item.isEnabled ? { toggleFilter(groupId: group.id, itemId: item.id) } : nil
                )
            }
        }
        .padding(.horizontal, WFMSpacing.l)
        .padding(.bottom, WFMSpacing.m)
    }

    // Список с чекбоксами для секций с >10 элементами
    private func checkboxContent(group: TaskFilterGroup) -> some View {
        VStack(spacing: 0) {
            ForEach(Array(group.items.enumerated()), id: \.element.id) { index, item in
                WFMSelectionCard(
                    title: item.title,
                    type: .select,
                    isChecked: item.isSelected,
                    showBorder: false,
                    onTap: item.isEnabled ? { toggleFilter(groupId: group.id, itemId: item.id) } : {}
                )
                .opacity(item.isEnabled ? 1 : 0.4)

                if index < group.items.count - 1 {
                    Divider()
                        .padding(.horizontal, WFMSpacing.l)
                }
            }
        }
        .padding(.horizontal, WFMSpacing.l)
        .padding(.bottom, WFMSpacing.m)
    }

    // MARK: - Bottom Buttons

    private var bottomButtons: some View {
        VStack(spacing: WFMSpacing.m) {
            WFMPrimaryButton(
                text: showButtonText,
                action: {
                    onApply(filterGroups)
                    onDismiss()
                }
            )
            .fixedSize(horizontal: false, vertical: true)

            Button("Очистить фильтры") {
                clearAllFilters()
            }
            .font(WFMTypography.headline12Medium)
            .foregroundStyle(
                hasAnySelected
                    ? colors.buttonLinkTextDefault
                    : colors.textBrandDisabled
            )
            .disabled(!hasAnySelected)
        }
        .padding(WFMSpacing.l)
    }

    // MARK: - Actions

    private func toggleFilter(groupId: String, itemId: String) {
        guard let gi = filterGroups.firstIndex(where: { $0.id == groupId }),
              let ii = filterGroups[gi].items.firstIndex(where: { $0.id == itemId }) else { return }
        filterGroups[gi].items[ii].isSelected.toggle()
    }

    private func deselectAllInGroup(groupId: String) {
        guard let gi = filterGroups.firstIndex(where: { $0.id == groupId }) else { return }
        for ii in filterGroups[gi].items.indices {
            filterGroups[gi].items[ii].isSelected = false
        }
    }

    private func clearAllFilters() {
        for gi in filterGroups.indices {
            for ii in filterGroups[gi].items.indices {
                filterGroups[gi].items[ii].isSelected = false
            }
        }
    }
}

// MARK: - Recompute Enabled State

/// Пересчитывает isEnabled для каждого элемента фильтра на основе taskFilterIndices.
/// Для группы G: enabled = индексы, встречающиеся в строках taskFilterIndices,
/// где все остальные группы с активными выборами совпадают.
func recomputeFilterEnabledState(filterGroups: [TaskFilterGroup], taskFilterIndices: [[Int]]) -> [TaskFilterGroup] {
    guard !taskFilterIndices.isEmpty else { return filterGroups }
    var updated = filterGroups
    for g in filterGroups.indices {
        var otherSelections: [Int: Set<Int>] = [:]
        for h in filterGroups.indices where h != g {
            let sel = Set(filterGroups[h].items.indices.filter { filterGroups[h].items[$0].isSelected })
            if !sel.isEmpty { otherSelections[h] = sel }
        }
        guard !otherSelections.isEmpty else {
            for i in updated[g].items.indices { updated[g].items[i].isEnabled = true }
            continue
        }
        var reachable = Set<Int>()
        for row in taskFilterIndices {
            guard row.count > g else { continue }
            guard otherSelections.allSatisfy({ h, selSet in h < row.count && selSet.contains(row[h]) }) else { continue }
            let idx = row[g]
            if idx >= 0 { reachable.insert(idx) }
        }
        for i in updated[g].items.indices { updated[g].items[i].isEnabled = reachable.contains(i) }
    }
    return updated
}

// MARK: - FlowLayout для чипов и тегов

private struct FiltersFlowLayout: Layout {
    var spacing: CGFloat = 8

    func sizeThatFits(proposal: ProposedViewSize, subviews: Subviews, cache: inout ()) -> CGSize {
        FlowResult(in: proposal.replacingUnspecifiedDimensions().width, subviews: subviews, spacing: spacing).size
    }

    func placeSubviews(in bounds: CGRect, proposal: ProposedViewSize, subviews: Subviews, cache: inout ()) {
        let result = FlowResult(in: bounds.width, subviews: subviews, spacing: spacing)
        for (index, subview) in subviews.enumerated() {
            subview.place(
                at: CGPoint(x: bounds.minX + result.frames[index].minX, y: bounds.minY + result.frames[index].minY),
                proposal: .unspecified
            )
        }
    }

    struct FlowResult {
        var size: CGSize = .zero
        var frames: [CGRect] = []

        init(in maxWidth: CGFloat, subviews: Subviews, spacing: CGFloat) {
            var x: CGFloat = 0
            var y: CGFloat = 0
            var lineHeight: CGFloat = 0

            for subview in subviews {
                let size = subview.sizeThatFits(.unspecified)
                if x + size.width > maxWidth, x > 0 {
                    x = 0
                    y += lineHeight + spacing
                    lineHeight = 0
                }
                frames.append(CGRect(x: x, y: y, width: size.width, height: size.height))
                lineHeight = max(lineHeight, size.height)
                x += size.width + spacing
            }
            self.size = CGSize(width: maxWidth, height: y + lineHeight)
        }
    }
}

// MARK: - Preview

private let previewFilterGroups: [TaskFilterGroup] = [
    TaskFilterGroup(
        id: "work_types",
        title: "Тип работ",
        items: [
            TaskFilterItem(id: "1", title: "Менеджерские задачи", isSelected: true),
            TaskFilterItem(id: "2", title: "Касса", isSelected: false),
            TaskFilterItem(id: "3", title: "КСО", isSelected: false),
            TaskFilterItem(id: "4", title: "Переоценка", isSelected: false),
            TaskFilterItem(id: "5", title: "Выкладка", isSelected: false),
            TaskFilterItem(id: "6", title: "Смена ценников", isSelected: false),
            TaskFilterItem(id: "7", title: "Другие работы", isSelected: false)
        ]
    ),
    TaskFilterGroup(
        id: "assignee_ids",
        title: "Сотрудники",
        items: [
            TaskFilterItem(id: "e1", title: "Алябьев А.Г.", isSelected: false),
            TaskFilterItem(id: "e2", title: "Глинка М.И.", isSelected: false),
            TaskFilterItem(id: "e3", title: "Мусоргский М.П.", isSelected: false)
        ]
    ),
    TaskFilterGroup(
        id: "zones",
        title: "Зона",
        items: [
            TaskFilterItem(id: "z1", title: "Алкоголь", isSelected: false),
            TaskFilterItem(id: "z2", title: "Бакалея", isSelected: false),
            TaskFilterItem(id: "z3", title: "Бытовая химия", isSelected: false),
            TaskFilterItem(id: "z4", title: "Бэк офис", isSelected: false),
            TaskFilterItem(id: "z5", title: "Заморозка", isSelected: false),
            TaskFilterItem(id: "z6", title: "ЗОЖ", isSelected: false),
            TaskFilterItem(id: "z7", title: "Молочка", isSelected: false),
            TaskFilterItem(id: "z8", title: "Мясо", isSelected: false),
            TaskFilterItem(id: "z9", title: "Овощи", isSelected: false),
            TaskFilterItem(id: "z10", title: "Рыба", isSelected: false),
            TaskFilterItem(id: "z11", title: "Хлеб", isSelected: false)
        ]
    )
]

private struct TaskFiltersBottomSheetPreview: View {
    @StateObject private var bottomSheetManager = BottomSheetManager()
    @Environment(\.wfmColors) private var colors

    var body: some View {
        VStack {
            Button("Открыть фильтры") {
                TaskFiltersBottomSheet.show(
                    bottomSheetManager: bottomSheetManager,
                    filterGroups: previewFilterGroups,
                    onApply: { groups in print("Applied filters: \(groups.flatMap { $0.items.filter(\.isSelected).map(\.title) })") }
                )
            }
        }
        .frame(maxWidth: .infinity, maxHeight: .infinity)
        .background(colors.surfaceBase)
        .wfmBottomSheet(
            isPresented: $bottomSheetManager.isPresented,
            showOverlay: bottomSheetManager.showOverlay
        ) {
            if let content = bottomSheetManager.content {
                content
            }
        }
    }
}

#Preview("Task Filters BottomSheet") {
    TaskFiltersBottomSheetPreview()
        .wfmTheme()
}

File: wfm-develop mobile 07.05.2026/wfm/mobile/android/app/src/main/java/com/beyondviolet/wfm/features/managertasks/TaskFiltersBottomSheet.kt
package com.beyondviolet.wfm.features.managertasks

import androidx.compose.animation.AnimatedVisibility
import androidx.compose.animation.expandVertically
import androidx.compose.animation.shrinkVertically
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.ExperimentalLayoutApi
import androidx.compose.foundation.layout.FlowRow
import androidx.compose.foundation.layout.PaddingValues
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material3.HorizontalDivider
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.material3.TextButton
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.graphicsLayer
import androidx.compose.ui.res.painterResource
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.ui.unit.dp
import com.beyondviolet.wfm.ui.components.BadgeColor
import com.beyondviolet.wfm.ui.components.WfmBadge
import com.beyondviolet.wfm.ui.components.WfmChip
import com.beyondviolet.wfm.ui.components.WfmChipState
import com.beyondviolet.wfm.ui.components.WfmFilterTag
import com.beyondviolet.wfm.ui.components.WfmPrimaryButton
import com.beyondviolet.wfm.ui.components.WfmSelectionCard
import com.beyondviolet.wfm.ui.components.WfmSelectionCardType
import com.beyondviolet.wfm.ui.theme.WfmRadius
import com.beyondviolet.wfm.ui.theme.WfmSpacing
import com.beyondviolet.wfm.ui.theme.WfmStroke
import com.beyondviolet.wfm.ui.theme.WfmTheme
import com.beyondviolet.wfm.ui.theme.WfmTypography
import com.beyondviolet.wfm.ui.theme.*

/**
 * Модель элемента фильтра для UI
 */
data class TaskFilterItem(
    val id: String,
    val title: String,
    val isSelected: Boolean,
    val isEnabled: Boolean = true
)

/**
 * Группа фильтров для UI
 */
data class TaskFilterGroup(
    val id: String,
    val title: String,
    val items: List<TaskFilterItem>
)

private const val FILTER_CHIP_THRESHOLD = 10

/**
 * Контент BottomSheet фильтров задач (тип работ, сотрудники, зоны)
 *
 * Используется внутри WfmBottomSheet:
 * ```
 * WfmBottomSheet(isVisible = showFilters, onDismiss = { showFilters = false }, showOverlay = true) {
 *     TaskFiltersBottomSheetContent(
 *         filterGroups = filterGroups,
 *         onFilterGroupsChange = { viewModel.updateFilters(it) },
 *         onApply = { viewModel.applyFilters(it) },
 *         onDismiss = { showFilters = false }
 *     )
 * }
 * ```
 */
@OptIn(ExperimentalLayoutApi::class)
@Composable
fun TaskFiltersBottomSheetContent(
    filterGroups: List<TaskFilterGroup>,
    taskFilterIndices: List<List<Int>> = emptyList(),
    onFilterGroupsChange: (List<TaskFilterGroup>) -> Unit,
    onApply: (List<TaskFilterGroup>) -> Unit,
    onDismiss: () -> Unit
) {
    val colors = WfmTheme.colors
    var expandedGroupId by remember { mutableStateOf<String?>(null) }

    val hasAnySelected = filterGroups.any { group -> group.items.any { it.isSelected } }
    val selectedGroups = filterGroups.filter { group -> group.items.any { it.isSelected } }

    // Количество задач, соответствующих текущему выбору фильтров
    val matchingTaskCount = remember(filterGroups, taskFilterIndices) {
        if (!hasAnySelected || taskFilterIndices.isEmpty()) return@remember -1
        val activeSelections = filterGroups.mapIndexedNotNull { g, group ->
            val selected = group.items.indices.filter { group.items[it].isSelected }.toSet()
            if (selected.isNotEmpty()) g to selected else null
        }.toMap()
        taskFilterIndices.count { row ->
            activeSelections.all { (g, selSet) -> g < row.size && row[g] in selSet }
        }
    }

    val showButtonText = if (matchingTaskCount >= 0) {
        "Показать ${matchingTaskCount} ${taskCountWord(matchingTaskCount)}"
    } else {
        "Показать задачи"
    }

    Column(modifier = Modifier.fillMaxWidth().verticalScroll(rememberScrollState())) {
        // Заголовок
        Box(
            modifier = Modifier
                .fillMaxWidth()
                .padding(horizontal = WfmSpacing.L)
                .padding(top = WfmSpacing.M, bottom = WfmSpacing.S)
                .height(44.dp),
            contentAlignment = Alignment.CenterStart
        ) {
            Text(
                text = "Фильтры",
                style = WfmTypography.Headline20Bold,
                color = colors.textPrimary
            )
        }

        // Секция "Вы выбрали" — показывается если хоть что-то выбрано
        if (selectedGroups.isNotEmpty()) {
            Column(
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(horizontal = WfmSpacing.L)
                    .padding(bottom = WfmSpacing.M),
                verticalArrangement = Arrangement.spacedBy(WfmSpacing.S)
            ) {
                Box(
                    modifier = Modifier
                        .fillMaxWidth()
                        .height(40.dp),
                    contentAlignment = Alignment.CenterStart
                ) {
                    Text(
                        text = "Вы выбрали",
                        style = WfmTypography.Headline16Bold,
                        color = colors.textPrimary
                    )
                }
                FlowRow(
                    horizontalArrangement = Arrangement.spacedBy(WfmSpacing.S),
                    verticalArrangement = Arrangement.spacedBy(WfmSpacing.S)
                ) {
                    selectedGroups.forEach { group ->
                        WfmFilterTag(
                            text = group.title,
                            onRemove = {
                                onFilterGroupsChange(deselectAllInGroup(filterGroups, group.id))
                            }
                        )
                    }
                }
            }
        }

        // Аккордеон секций
        filterGroups.forEachIndexed { index, group ->
            val isExpanded = expandedGroupId == group.id
            val selectedCount = group.items.count { it.isSelected }

            Column(modifier = Modifier.fillMaxWidth()) {
                // Заголовок-кнопка аккордеона
                Row(
                    modifier = Modifier
                        .fillMaxWidth()
                        .clickable {
                            if (isExpanded) {
                                expandedGroupId = null
                            } else {
                                // Пересчитываем enabled перед открытием секции
                                onFilterGroupsChange(recomputeFilterEnabledState(filterGroups, taskFilterIndices))
                                expandedGroupId = group.id
                            }
                        }
                        .padding(horizontal = WfmSpacing.L, vertical = WfmSpacing.S),
                    verticalAlignment = Alignment.CenterVertically,
                    horizontalArrangement = Arrangement.spacedBy(WfmSpacing.XXS)
                ) {
                    Text(
                        text = group.title,
                        style = WfmTypography.Headline16Bold,
                        color = colors.textPrimary
                    )

                    if (selectedCount > 0) {
                        WfmBadge(text = "$selectedCount", color = BadgeColor.VIOLET)
                    }

                    Spacer(modifier = Modifier.weight(1f))

                    if (selectedCount > 0) {
                        TextButton(
                            onClick = {
                                onFilterGroupsChange(deselectAllInGroup(filterGroups, group.id))
                            },
                            contentPadding = PaddingValues(horizontal = WfmSpacing.XS, vertical = 0.dp)
                        ) {
                            Text(
                                text = "Очистить",
                                style = WfmTypography.Headline12Medium,
                                color = colors.buttonLinkTextDefault
                            )
                        }
                    }

                    Icon(
                        painter = painterResource(id = if (isExpanded) com.beyondviolet.wfm.ui.R.drawable.ic_chevron_up else com.beyondviolet.wfm.ui.R.drawable.ic_chevron_down),
                        contentDescription = null,
                        tint = colors.iconPrimary,
                        modifier = Modifier.size(20.dp)
                    )
                }

                // Контент секции: чипы (≤10) или список с чекбоксами (>10)
                AnimatedVisibility(
                    visible = isExpanded,
                    enter = expandVertically(),
                    exit = shrinkVertically()
                ) {
                    if (group.items.size <= FILTER_CHIP_THRESHOLD) {
                        // Чипы
                        FlowRow(
                            modifier = Modifier
                                .fillMaxWidth()
                                .padding(horizontal = WfmSpacing.L)
                                .padding(bottom = WfmSpacing.M),
                            horizontalArrangement = Arrangement.spacedBy(WfmSpacing.S),
                            verticalArrangement = Arrangement.spacedBy(WfmSpacing.S)
                        ) {
                            group.items.forEach { item ->
                                WfmChip(
                                    text = item.title,
                                    state = when {
                                        !item.isEnabled -> WfmChipState.DISABLED
                                        item.isSelected -> WfmChipState.ACTIVE
                                        else -> WfmChipState.DEFAULT
                                    },
                                    onClick = if (item.isEnabled) {
                                        { onFilterGroupsChange(toggleFilterItem(filterGroups, group.id, item.id)) }
                                    } else null
                                )
                            }
                        }
                    } else {
                        // Список с чекбоксами
                        Column(
                            modifier = Modifier
                                .fillMaxWidth()
                                .padding(horizontal = WfmSpacing.L)
                                .padding(bottom = WfmSpacing.M)
                                .clip(RoundedCornerShape(WfmRadius.XL))
                                .background(colors.surfaceSecondary)
                                .border(
                                    width = WfmStroke.S,
                                    color = colors.borderSecondary,
                                    shape = RoundedCornerShape(WfmRadius.XL)
                                )
                        ) {
                            group.items.forEachIndexed { itemIndex, item ->
                                androidx.compose.foundation.layout.Box(
                                    modifier = Modifier.graphicsLayer { alpha = if (item.isEnabled) 1f else 0.4f }
                                ) {
                                    WfmSelectionCard(
                                        title = item.title,
                                        type = WfmSelectionCardType.SELECT,
                                        isChecked = item.isSelected,
                                        showBorder = false,
                                        onTap = if (item.isEnabled) {
                                            { onFilterGroupsChange(toggleFilterItem(filterGroups, group.id, item.id)) }
                                        } else {
                                            {}
                                        }
                                    )
                                }
                                if (itemIndex < group.items.lastIndex) {
                                    HorizontalDivider(
                                        modifier = Modifier.padding(horizontal = WfmSpacing.M),
                                        color = colors.borderSecondary
                                    )
                                }
                            }
                        }
                    }
                }

                if (index < filterGroups.lastIndex) {
                    HorizontalDivider(
                        modifier = Modifier.padding(horizontal = WfmSpacing.L),
                        color = colors.borderSecondary
                    )
                }
            }
        }

        HorizontalDivider(color = colors.borderSecondary)

        // Кнопки внизу
        Column(
            modifier = Modifier
                .fillMaxWidth()
                .padding(WfmSpacing.L),
            verticalArrangement = Arrangement.spacedBy(WfmSpacing.M),
            horizontalAlignment = Alignment.CenterHorizontally
        ) {
            WfmPrimaryButton(
                text = showButtonText,
                onClick = {
                    onApply(filterGroups)
                    onDismiss()
                },
                modifier = Modifier.fillMaxWidth()
            )

            TextButton(
                onClick = { onFilterGroupsChange(clearAllFilters(filterGroups)) },
                enabled = hasAnySelected,
                contentPadding = PaddingValues(horizontal = WfmSpacing.S, vertical = WfmSpacing.XXS)
            ) {
                Text(
                    text = "Очистить фильтры",
                    style = WfmTypography.Headline12Medium,
                    color = if (hasAnySelected) {
                        colors.buttonLinkTextDefault
                    } else {
                        colors.textBrandDisabled
                    }
                )
            }
        }
    }
}

// MARK: - Helpers

/**
 * Пересчитывает isEnabled для каждого элемента фильтра на основе taskFilterIndices.
 * Для группы G: enabled = индексы, встречающиеся в строках taskFilterIndices,
 * где все остальные группы с активными выборами совпадают.
 */
fun recomputeFilterEnabledState(
    groups: List<TaskFilterGroup>,
    indices: List<List<Int>>
): List<TaskFilterGroup> {
    if (indices.isEmpty()) return groups
    return groups.mapIndexed { g, group ->
        val otherSelections = groups.indices
            .filter { it != g }
            .mapNotNull { h ->
                val sel = groups[h].items.indices.filter { groups[h].items[it].isSelected }.toSet()
                if (sel.isNotEmpty()) h to sel else null
            }
            .toMap()
        if (otherSelections.isEmpty()) {
            return@mapIndexed group.copy(items = group.items.map { it.copy(isEnabled = true) })
        }
        val reachable = mutableSetOf<Int>()
        for (row in indices) {
            if (row.size <= g) continue
            if (!otherSelections.all { (h, selSet) -> h < row.size && row[h] in selSet }) continue
            val idx = row[g]
            if (idx >= 0) reachable.add(idx)
        }
        group.copy(items = group.items.mapIndexed { i, item -> item.copy(isEnabled = i in reachable) })
    }
}

private fun taskCountWord(count: Int): String {
    val mod100 = count % 100
    val mod10 = count % 10
    return when {
        mod100 in 11..14 -> "задач"
        mod10 == 1 -> "задачу"
        mod10 in 2..4 -> "задачи"
        else -> "задач"
    }
}

private fun toggleFilterItem(
    groups: List<TaskFilterGroup>,
    groupId: String,
    itemId: String
): List<TaskFilterGroup> = groups.map { group ->
    if (group.id != groupId) group
    else group.copy(items = group.items.map { item ->
        if (item.id == itemId) item.copy(isSelected = !item.isSelected) else item
    })
}

private fun deselectAllInGroup(
    groups: List<TaskFilterGroup>,
    groupId: String
): List<TaskFilterGroup> = groups.map { group ->
    if (group.id != groupId) group
    else group.copy(items = group.items.map { it.copy(isSelected = false) })
}

private fun clearAllFilters(groups: List<TaskFilterGroup>): List<TaskFilterGroup> =
    groups.map { group ->
        group.copy(items = group.items.map { it.copy(isSelected = false) })
    }

// MARK: - Preview

@OptIn(ExperimentalLayoutApi::class)
@Preview(showBackground = true, backgroundColor = 0xFFFFFF)
@Composable
private fun TaskFiltersBottomSheetPreview() {
    WfmTheme {
        val groups = listOf(
            TaskFilterGroup(
                id = "work_types",
                title = "Тип работ",
                items = listOf(
                    TaskFilterItem(id = "1", title = "Менеджерские задачи", isSelected = true),
                    TaskFilterItem(id = "2", title = "Касса", isSelected = false),
                    TaskFilterItem(id = "3", title = "КСО", isSelected = false),
                    TaskFilterItem(id = "4", title = "Переоценка", isSelected = false),
                    TaskFilterItem(id = "5", title = "Выкладка", isSelected = false),
                    TaskFilterItem(id = "6", title = "Смена ценников", isSelected = false),
                    TaskFilterItem(id = "7", title = "Другие работы", isSelected = false)
                )
            ),
            TaskFilterGroup(
                id = "employees",
                title = "Сотрудники",
                items = listOf(
                    TaskFilterItem(id = "e1", title = "Алябьев А.Г.", isSelected = false),
                    TaskFilterItem(id = "e2", title = "Глинка М.И.", isSelected = false),
                    TaskFilterItem(id = "e3", title = "Мусоргский М.П.", isSelected = false)
                )
            ),
            TaskFilterGroup(
                id = "zones",
                title = "Зона",
                items = listOf(
                    TaskFilterItem(id = "z1", title = "Алкоголь", isSelected = false),
                    TaskFilterItem(id = "z2", title = "Бакалея", isSelected = false),
                    TaskFilterItem(id = "z3", title = "Бытовая химия", isSelected = false)
                )
            )
        )

        var filterGroups by remember { mutableStateOf(groups) }

        TaskFiltersBottomSheetContent(
            filterGroups = filterGroups,
            onFilterGroupsChange = { filterGroups = it },
            onApply = {},
            onDismiss = {}
        )
    }
}
